packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx

```diff
@@ -423,15 +515,127 @@ export class ScrollablePaneBase extends BaseComponent<IScrollablePaneProps, IScr
     return contentContainer ? contentContainer.offsetHeight - contentContainer.clientHeight : 0;
   }
 
+  private _getScrollbarHeightFromLocalStorage(): number | undefined {
+    if (typeof window !== 'undefined' && 'localStorage' in window) {
+      const scrollbarHeight = Number(window.localStorage.getItem(ScrollablePaneBase._scrollbarHeightKey));
+      if (!!scrollbarHeight && !isNaN(scrollbarHeight)) {
+        return scrollbarHeight;
+      }
+    }
+    return undefined;
+  }
+
+  private _storeScrollbarHeightInLocalStorage(scrollbarHeight: number) {
+    if (typeof window !== 'undefined' && 'localStorage' in window) {
+      window.localStorage.setItem(ScrollablePaneBase._scrollbarHeightKey, scrollbarHeight.toString());
+    }
+  }
+
   private _onScroll = () => {
     const { contentContainer } = this;
 
     if (contentContainer) {
-      this._stickies.forEach((sticky: Sticky) => {
-        sticky.syncScroll(contentContainer);
+      // sync Sticky scroll if contentContainer has scrolled horizontally
+      if (this._scrollLeft !== contentContainer.scrollLeft) {
+        this._scrollLeft = contentContainer.scrollLeft;
+        this._stickies.forEach((sticky: Sticky) => {
+          sticky.syncScroll(contentContainer);
+        });
+      }
+      if (this._scrollTop !== contentContainer.scrollTop) {
+        this._userInteractionStarted = true;
+        this._scrollTop = contentContainer.scrollTop;
+        this._notifyThrottled();
+      }
+    }
+  };
+
+  private _sortBasedOnOrder(): boolean {
+    return !!this.props.optimizeForPerformace;
+  }
+
+  private _stickyContainerContainsStickyContent(sticky: Sticky, stickyPositionType: StickyPositionType): boolean {
+    return stickyPositionType === StickyPositionType.Header
+      ? !!this.stickyAbove && !!sticky.stickyContentTop && this.stickyAbove.contains(sticky.stickyContentTop)
+      : !!this.stickyBelow && !!sticky.stickyContentBottom && this.stickyBelow.contains(sticky.stickyContentBottom);
+  }
+
+  private _sortStickyList(stickyList: Sticky[], sortBasedOnOrder: boolean): Sticky[] {
```

Comment:
Can `sortBasedOnOrder` take a type instead? Could be

    type _SortKey = 'order' | 'distanceFromTop';
    private _sortStickyList(stickyList: Sticky[], sortBasis: _SortKey): Sticky[]

This way it avoids putting a boolean in the list of parameters.